src/OpenTelemetry.Instrumentation.GrpcNetClient/TracerProviderBuilderExtensions.cs

```diff
@@ -28,16 +28,21 @@ public static class TracerProviderBuilderExtensions
         /// Enables gRPClient Instrumentation.
         /// </summary>
         /// <param name="builder"><see cref="TracerProviderBuilder"/> being configured.</param>
+        /// <param name="configureGrpcClientInstrumentationOptions">GrpcClient configuration options.</param>
         /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
         public static TracerProviderBuilder AddGrpcClientInstrumentation(
```

Comment:
@cijothomas @CodeBlanch this makes me wonder if we want `AddInstrumentation<GrpcClientInstrumentation>(options)`

Reply:
I like the idea but I see some challenges.

What we want to do:
```csharp
AddInstrumentation<GrpcClientInstrumentation>(options => options.Something = true);
```

Probably has to be more like:
```csharp
AddInstrumentation<GrpcClientInstrumentation, GrpcClientInstrumentationOptions>(options => options.Something = true);
```

Right there we lose a lot of value. The user has to know the instrumentation type and the options type.

What the method has to do:
```csharp
        public static TracerProviderBuilder AddInstrumentation<TInstrumentation, TOptions>(
            this TracerProviderBuilder builder,
            Action<TOptions> configure = null)
            where TOptions : new()
        {
            TOptions options = new TOptions();
            configure?.Invoke(options);

            builder.AddInstrumentation((activitySource) => new TInstrumentation(activitySource, options));

            return builder;
        }
```

The `new TInstrumentation(activitySource, options)` won't compile. We'll have to do a factory or use reflection kind of like how ASP.NET Core does middleware. I think that would be worth pursuing but having to specify the two types gives me pause.